build_tools/fetch_artifacts.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+# This script provides a somewhat dynamic way to
+# retrieve artifacts from s3
+
+import argparse
+import subprocess
+import sys
+
+GENERIC_VARIANT = "generic"
+
+
+def log(*args, **kwargs):
+    print(*args, **kwargs)
+    sys.stdout.flush()
+
+
+def s3_exec(variant, package, run_id, build_dir):
+    cmd = [
+        "aws",
+        "s3",
+        "cp",
+        f"s3://therock-artifacts/{run_id}/{package}_{variant}.tar.xz",
+        build_dir,
+        "--no-sign-request",
+    ]
+    log(f"++ Exec [{cmd}]")
+    subprocess.run(cmd)
+
+
+def retrieve_base_artifacts(run_id, build_dir):
+    base_artifacts = [
+        "core-runtime_run",
+        "core-runtime_lib",
+        "sysdeps_lib",
+        "base_lib",
+        "amd-llvm_run",
+        "amd-llvm_lib",
+        "core-hip_lib",
+        "core-hip_dev",
+        "host-blas_lib",
```

Comment:
I think host-blas is only needed together with the blas artifacts so I would move it there.